crates/humanode-runtime/src/lib.rs

```diff
@@ -568,64 +570,12 @@ impl pallet_humanode_session::Config for Runtime {
     type MaxBioauthValidators = <Runtime as pallet_bioauth::Config>::MaxAuthentications;
 }
 
-pub struct OffenceSlasher;
-
-impl
-    sp_staking::offence::OnOffenceHandler<
-        AccountId,
-        <Runtime as pallet_offences::Config>::IdentificationTuple,
-        Weight,
-    > for OffenceSlasher
-{
-    fn on_offence(
-        offenders: &[sp_staking::offence::OffenceDetails<
-            AccountId,
-            <Runtime as pallet_offences::Config>::IdentificationTuple,
-        >],
-        _slash_fraction: &[Perbill],
-        _session: sp_staking::SessionIndex,
-        disable_strategy: sp_staking::offence::DisableStrategy,
-    ) -> Weight {
-        if disable_strategy == sp_staking::offence::DisableStrategy::Never {
-            return Weight::zero();
-        }
-        let mut weight: Weight = Weight::zero();
-        let weights = <Runtime as frame_system::Config>::DbWeight::get();
-        let mut should_be_deauthenticated = Vec::with_capacity(offenders.len());
-        for details in offenders {
-            let (_offender, identity) = &details.offender;
-            match identity {
-                pallet_humanode_session::Identification::Bioauth(authentication) => {
-                    should_be_deauthenticated.push(authentication.public_key.clone());
-                }
-                pallet_humanode_session::Identification::Bootnode(..) => {
-                    // Never slash the bootnodes.
-                }
-            }
-        }
-        if !should_be_deauthenticated.is_empty() {
-            let deauthenticated_public_keys =
-                Bioauth::deauthenticate(should_be_deauthenticated, DeauthenticationReason::Offence);
-            weight = weight.saturating_add(
-                weights.reads_writes(
-                    1,
-                    deauthenticated_public_keys
-                        .len()
-                        .try_into()
-                        .expect("casting usize to u64 never fails in 64bit and 32bit word cpus"),
-                ),
-            );
-        }
-        weight
-    }
-}
-
 impl pallet_im_online::Config for Runtime {
     type AuthorityId = ImOnlineId;
     type RuntimeEvent = RuntimeEvent;
     type NextSessionRotation = Babe;
     type ValidatorSet = Historical;
-    type ReportUnresponsiveness = Offences;
+    type ReportUnresponsiveness = OffenceHandler<Offences>;
```

Comment:
Do we need pallet offences at all?

Reply:
Good question, we don't use any logic provided by this pallet. So, we can easily remove it now with own offence handler, I think.

Reply:
Done